backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/exceptions/DocumentFieldReadonlyException.java

```diff
@@ -30,13 +37,16 @@
 @ResponseStatus(code = HttpStatus.FORBIDDEN)
 public class DocumentFieldReadonlyException extends AdempiereException
 {
-	public DocumentFieldReadonlyException(final String fieldName, final Object value)
+	public static final AdMessageKey CHANGING_READONLY_FIELD_NOT_ALLOWED_ERROR = AdMessageKey.of("de.metas.ui.web.window.exceptions.DocumentFieldReadonlyException.Msg");
+
+	public DocumentFieldReadonlyException(@NonNull final String fieldName, @Nullable final Object value)
 	{
 		super(buildMsg(fieldName, value));
 	}
 
-	private static String buildMsg(final String fieldName, final Object value)
+	private static ITranslatableString buildMsg(@NonNull final String fieldName, @Nullable final Object value)
 	{
-		return "Changing " + fieldName + " to '" + value + "' is not allowed because the field is readonly";
+		return Services.get(IMsgBL.class).getTranslatableMsgText(CHANGING_READONLY_FIELD_NOT_ALLOWED_ERROR, fieldName, value);
 	}
```

Comment:
I would simplify it. (Don't forget to remove unused imports after)
```java
	public DocumentFieldReadonlyException(@NonNull final String fieldName, @Nullable final Object value)
	{
		super(CHANGING_READONLY_FIELD_NOT_ALLOWED_ERROR, fieldName, value);
	}
```

Reply:
methode simplified